addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -83,8 +83,10 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
 
     async scanAndUseGiftCard() {
       let giftCard = await this.getGiftCard();
-      if (!giftCard) return;
-
+      if (!giftCard) {
```

Comment:
Not sure about the error message, currently the widget just do nothing and it might be intentional, we'll check with the POS team